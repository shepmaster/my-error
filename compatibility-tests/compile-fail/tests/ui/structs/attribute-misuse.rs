mod unknown_attributes {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(unknown_attribute)]
    struct Error {}
}

mod invalid_on_struct {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(source(true))]
    #[snafu(backtrace)]
    struct Error {}
}

mod invalid_on_field {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    struct Error {
        #[snafu(display("display should not work here"))]
        #[snafu(visibility(pub))]
        #[snafu(context)]
        id: i32,
    }
}

fn main() {}
