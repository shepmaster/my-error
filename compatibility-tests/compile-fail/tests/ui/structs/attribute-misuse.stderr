error: expected `display`, `visibility`, `source`, `backtrace`, `context`, or `crate_root`
 --> $DIR/attribute-misuse.rs:5:13
  |
5 |     #[snafu(unknown_attribute)]
  |             ^^^^^^^^^^^^^^^^^

error: `source` attribute is only valid on enum variant or struct fields with a name, not on a named struct
  --> $DIR/attribute-misuse.rs:13:13
   |
13 |     #[snafu(source(true))]
   |             ^^^^^^^^^^^^

error: `backtrace` attribute is only valid on enum variant or struct fields with a name, not on a named struct
  --> $DIR/attribute-misuse.rs:14:13
   |
14 |     #[snafu(backtrace)]
   |             ^^^^^^^^^

error: `display` attribute is only valid on enum variants or structs with named fields, not on a field
  --> $DIR/attribute-misuse.rs:23:17
   |
23 |         #[snafu(display("display should not work here"))]
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `visibility` attribute is only valid on an enum, enum variants, or a struct with named fields, not on a field
  --> $DIR/attribute-misuse.rs:24:17
   |
24 |         #[snafu(visibility(pub))]
   |                 ^^^^^^^^^^^^^^^

error: `context` attribute is only valid on enum variants or structs with named fields, not on a field
  --> $DIR/attribute-misuse.rs:25:17
   |
25 |         #[snafu(context)]
   |                 ^^^^^^^
